src/font/lv_font_cache.c

```diff
@@ -0,0 +1,240 @@
+static void font_bitmap_free_cb(font_bitmap_cache_data_t * data, void * user_data);
+static lv_cache_compare_res_t font_bitmap_compare_cb(const font_bitmap_cache_data_t * lhs,
+                                                     const font_bitmap_cache_data_t * rhs);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void lv_font_cache_init(uint32_t max_glyph_cnt)
+{
+    LV_ASSERT_MSG(font_bitmap_cache == NULL, "Font cache already initialized");
+
+    lv_cache_ops_t ops = {
+        .compare_cb = (lv_cache_compare_cb_t)font_bitmap_compare_cb,
+        .create_cb = (lv_cache_create_cb_t)font_bitmap_create_cb,
+        .free_cb = (lv_cache_free_cb_t)font_bitmap_free_cb,
+    };
+
+    lv_cache_t * cache = lv_cache_create(&lv_cache_class_lru_rb_count, sizeof(font_bitmap_cache_data_t),
+                                         max_glyph_cnt, ops);
+    lv_cache_set_name(cache, "FONT_BITMAP");
+
+    font_bitmap_cache = cache;
+}
+
+void lv_font_cache_deinit(void)
+{
+    lv_cache_destroy(font_bitmap_cache, NULL);
+    font_bitmap_cache = NULL;
+}
+
+const void * lv_font_cache_get_glyph_bitmap(lv_font_glyph_dsc_t * g_dsc, lv_draw_buf_t * draw_buf)
+{
+    LV_ASSERT_NULL(g_dsc);
+
+    const lv_font_t * font = g_dsc->resolved_font;
+
+    /**
+     * If the font image buffer is provided externally, or it is not a built-in font,
+     * use the original function to get the bitmap.
+     */
+    if(draw_buf || !has_static_bitmap(font)) {
+        return font->get_glyph_bitmap(g_dsc, draw_buf);
+    }
+
+    const lv_font_fmt_txt_dsc_t * fdsc = (lv_font_fmt_txt_dsc_t *)font->dsc;
+    const uint32_t gid = g_dsc->gid.index;
+    if(!gid) return NULL;
+
+    const lv_font_fmt_txt_glyph_dsc_t * gdsc = &fdsc->glyph_dsc[gid];
+    const uint8_t * raw_glyph_bitmap = &fdsc->glyph_bitmap[gdsc->bitmap_index];
+
+    if(g_dsc->req_raw_bitmap) return raw_glyph_bitmap;
+
+    const uint32_t gsize = gdsc->box_w * gdsc->box_h;
+    if(gsize == 0) return NULL;
+
+    /**
+     * To prevent multithreaded access to the font_static_bitmap_draw_buf global variable,
+     * only non-OS mode is supported
+     */
+#if LV_USE_OS == LV_OS_NONE
+    /* If the alignment, stride, and color format is correct, bypass the cache */
+    if((g_dsc->format == LV_FONT_GLYPH_FORMAT_A8 || g_dsc->format == LV_FONT_GLYPH_FORMAT_A8_ALIGNED)
+       && raw_glyph_bitmap == lv_draw_buf_align((void *)raw_glyph_bitmap, LV_COLOR_FORMAT_A8)
+       && g_dsc->box_w == lv_draw_buf_width_to_stride(g_dsc->box_w, LV_COLOR_FORMAT_A8)) {
+
+        const uint32_t a8_stride = gdsc->box_w * sizeof(uint8_t);
+
+        /* Use the static bitmap draw buffer to avoid memory allocation */
+        lv_draw_buf_init(
+            &font_static_bitmap_draw_buf,
```

Comment:
@kisvegabor Is there any multi-threaded competition risk for this `font_static_bitmap_draw_buf` global variable?

Reply:
I will remove it temporarily and add it again when necessary.